relay-server/src/services/processor/span/processing.rs

```diff
@@ -141,6 +156,37 @@ pub fn process(
 
         ItemAction::Keep
     });
+
+    for mut transaction in extracted_transactions {
+        // Give each transaction event a new random ID:
+        transaction.id = EventId::new().into();
+
+        // Enqueue a full processing request for every extracted transaction item.
+        match Envelope::try_from_event(state.envelope().headers().clone(), transaction) {
+            Ok(mut envelope) => {
+                // We don't want to extract spans or span metrics from a transaction extracted from spans,
+                // so set the spans_extracted flag:
+                for item in envelope.items_mut() {
+                    item.set_spans_extracted(true);
+                }
+
+                addrs.envelope_processor.send(ProcessEnvelope {
+                    envelope: ManagedEnvelope::standalone(
+                        envelope,
+                        addrs.outcome_aggregator.clone(),
+                        addrs.test_store.clone(),
+                        ProcessingGroup::Transaction,
+                    ),
+                    project_state: state.project_state.clone(),
+                    sampling_project_state: state.sampling_project_state.clone(),
+                    reservoir_counters: state.reservoir.counters().clone(),
```

Comment:
Are we cloning reservoir counters twice? `counters()` clones once and there's an explicit clone here. `counters` should return a reference and we should clone here instead.